backend/main.py

```
@@ -475,6 +462,54 @@ async def get_semantic_hierarchy(request: Request):
+            "prompt": prompt,
+            "task_id": task_id,
+            "generated_at": datetime.now().isoformat(),
+            "prompt_length": len(prompt),
+            "related_memories_count": len(related_memories)
+        }
+        
+    except Exception as e:
+        logger.error(f"Error generating prompt for task {task_id}: {e}")
+        import traceback
+        logger.error(f"Traceback: {traceback.format_exc()}")
+        raise HTTPException(status_code=500, detail="Failed to generate prompt")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)